Directory.Build.props

```diff
@@ -327,6 +326,15 @@
     <EnableMultiFileILCTests Condition="'$(ConfigurationGroup)|$(ArchGroup)' == 'Release|x64' Or '$(ConfigurationGroup)|$(ArchGroup)' == 'Debug|x86' Or '$(ConfigurationGroup)|$(ArchGroup)' == 'Release|arm'">true</EnableMultiFileILCTests> -->
   </PropertyGroup>
 
+  <!-- Specify the default constants for each target group -->
+  <PropertyGroup>
+    <DefineConstants Condition="'$(TargetsUap)' == 'true'">$(DefineConstants);uap</DefineConstants>
+    <DefineConstants Condition="'$(TargetsNetFx)' == 'true'">$(DefineConstants);netfx</DefineConstants>
```

Comment:
This condition isn't the same as `'$(TargetGroup)'=='netfx'`, as TargetsNetFx is broader. It maybe OK but we just need to ensure that the conditions you replaced match this.

Reply:
I was quite convinced that is it ok, but I can change the condition if you prefer.

Reply:
I don't like the inconsistency between these so I'd prefer we make them all of the pattern `'$(TargetGroup)'=='netfx'` so that it is clear that you get the define of what you are targeting. We can add more defines if necessary in the future.

Reply:
I am not big fan of the inconsistency either, which is why I started off with a version that used `Condition="'$(Targets[Uap/NetFx/NetCoreApp/...)"` everywhere. Unfortunately that changed the meaning of "netcoreapp", which was obviously not ok since it's used as "current .NET Core version", not "any .NET Core version". However, "netfx" always had the meaning "any .NET Framework version".

Reply:
>However, "netfx" always had the meaning "any .NET Framework version".

I don't think that is true as it means netfx latest which is currently mapped to net472 but it changes overtime. Sorry to push on this more but I would still like these conditions to be updated to match exact TargetGroup instead of conditions that imply more. I would like to see:

```
    <DefineConstants Condition="'$(TargetGroup)' == 'uap'">$(DefineConstants),uap</DefineConstants>
    <DefineConstants Condition="'$(TargetGroup)' == 'netfx'">$(DefineConstants),netfx</DefineConstants>
    <DefineConstants Condition="'$(TargetGroup)' == 'netstandard'">$(DefineConstants),netstandard</DefineConstants>
```

This makes this repo consistent and this is also similar to how it works in the shipping SDK where one target equates to one define and not a family of defines.

Reply:
I'll recheck it. Things have changed a bit since I started it the PR. Looks like 'netfx' and 'uap' is safe to change now. 'netstandard' is currently used only on one place as far as I can tell, so it may be safe too.

Reply:
Let's see if the build turns green.